### ISSUE TITLE
No audio stream present. Audio cannot be recorded.

### ISSUE THREAD

**Issue by traeanto** [ammen99/wf-recorder#211]
Im using pipewire-pulse for my audio. Here is my output:

![image](https://user-images.githubusercontent.com/126303616/222320516-17641409-d584-40c5-8ce9-d37c856307bd.png)

![image](https://user-images.githubusercontent.com/126303616/222320338-a3ab8cde-49d0-4ed8-b406-df1f212c0be4.png)

**Comment by traeanto** [comment]
I have tried to record a video but the audio is not recorded properly (just silent video). I have also tried recording audio with ffmpeg, and it works fine (but can't record video in wayland).
i just want to record video with audio~

**Comment by soreau** [comment]
What happens if you try `wf-recorder -a -f record.mkv`?

**Comment by traeanto** [comment]
UPDATE: when recording video with `wf-recorder -a -f record.mkv`, audio can also be recorded IF I remove the headset connected to the laptop. BUT when I try to play the video, the sound only comes from the left side of the speaker or headset. That's weird~

**Comment by Selmer443** [comment]
getting the same issue but dont see an invalid audio stream error. Just no audio plays when I playback the recording

**Comment by soreau** [comment]
As the wf-recorder man page states:

"To specify an audio device, use the --a<DEVICE> or ---audio=<DEVICE> options."

This means that `-a <DEVICE>` (with a space between) will still choose the default device.

**Comment by Selmer443** [comment]
> As the wf-recorder man page states:
> 
> 
> 
> "To specify an audio device, use the -a\<DEVICE\> or --audio=\<DEVICE\> options."
> 
> 
> 
> This means that `-a <DEVICE>` (with a space between) will still choose the default device.

I've tried both of these, still no audio is coming through in playback.

**Comment by soreau** [comment]
@Selmer443 Maybe check the build log to make sure wf-recorder is built with pulseaudio support? What device name are you trying to record from?

**Comment by Selmer443** [comment]
I use this to get the default output device pactl get-default-sink

**Comment by soreau** [comment]
It might need `.monitor` appended or something, try `pactl list sources | grep Name`.

**Comment by Selmer443** [comment]
yeah, it needed .monitor at the end. weird because before it was able to pick it up, it might be pactl was returning the .monitor appended to the default sink before, but that changed on my system for some reason. Not really sure, i'll have to look into that. thanks for the help

**Comment by mrsteve0924** [comment]
i am running into this same issue when using bluetooth headset.  i have use --audio=devicename.monitor

Seems the -a flag works only when my speakers are the default device but not when the bluetooth headset is the default device. 

would be nice if it could work with bluetooth as the default device without having to specify it
